api/src/users/services/users.service.ts

```diff
@@ -1,85 +1,140 @@
+            })
+            .catch(error => {
+                throw error;
+            });
 
         if (user) throw new ConflictException("Duplicated username");
 
         return this.userRepository.save(userCredentialsDto);
     }
 
     async findAll() {
-        return this.userRepository.find({
-            relations: {
-                userAvatar: true
-            }
-        });
+        return this.userRepository.find();
     }
 
     async findOne(id: number) {
-        return this.userRepository
-            .findOneOrFail({
-                where: { id },
-                relations: { userAvatar: true }
-            })
-            .catch(() => {
+        return this.userRepository.findOneByOrFail({ id }).catch(error => {
+            if (error instanceof EntityNotFoundError)
                 throw new NotFoundException();
+
+            throw error;
+        });
+    }
+
+    async findOneByIdAndUserJwtId(id: number, userId: number) {
+        const user = await this.userRepository
+            .findOneByOrFail({ id })
+            .catch(error => {
+                if (error instanceof EntityNotFoundError)
+                    throw new NotFoundException();
+
+                throw error;
             });
+
+        if (!this.isDevelopment && user.id !== userId)
+            throw new ForbiddenException("You are not the owner of account");
+
+        return user;
     }
 
     async findOneByUsername(username: string) {
-        return this.userRepository.findOneByOrFail({ username }).catch(() => {
-            throw new NotFoundException();
-        });
+        return this.userRepository
+            .findOneByOrFail({ username })
+            .catch(error => {
+                if (error instanceof EntityNotFoundError)
+                    throw new NotFoundException();
+
+                throw error;
+            });
     }
 
     async selectPassword(username: string) {
         return this.userRepository
             .findOneOrFail({
                 where: { username },
-                select: ["password"]
+                select: ["id", "password"]
             })
-            .catch(() => {
-                throw new NotFoundException();
+            .catch(error => {
+                if (error instanceof EntityNotFoundError)
+                    throw new NotFoundException("Username don't exist");
+
+                throw error;
             });
     }
 
     async delete(id: number) {
-        const result = (await this.userRepository.delete(id)).affected;
-
-        if (!result) throw new NotFoundException();
+        return this.userRepository.delete(id);
     }
 
-    async update(id: number, updateUserDto: UpdateUserDto) {
+    async update(
+        id: number,
+        userId: number,
+        updateUserDto: UpdateUserDto,
+        file: IUploadedFile
+    ) {
+        const user = await this.findOneByIdAndUserJwtId(id, userId);
+
         updateUserDto.username = updateUserDto.username
```

Comment:
This is already replaced in DTO.

Reply:
Ye, you're right, i leave it here, because something didn't work ealier but now it's working, it's issue with Nest and this decorator in DTO, but it's all right now ;)